### PR TITLE
Use the same set of compiler options between dune and ocamlbuild

### PR DESCRIPTION
**Pull request by kit-ty-kate**
As directed in https://github.com/dbuenzli/cmdliner/pull/159#issuecomment-1352973481

### CODE REVIEW

File: dune [hunk 1]
```diff
@@ -0,0 +1 @@
+(env (_ (flags -bin-annot -safe-string))) ; Use the same flags as with ocamlbuild
```
Comment (kit-ty-kate) [inline dune:1]: side note: `-bin-annot` is optional (as it is always added to the flags) but it reflects ocamlbuild's `_tags` file more strictly if we add it by hand. But it's up to you, whatever you prefer.

Comment (dbuenzli) [reply dune:1]: We don't need to reflect ocamlbuild's `_tags` file the default build is `Makefile` based. We just need to reflect the standard OCaml flags.

File: src/dune [hunk 2]
```diff
@@ -1,4 +1,3 @@
 (library
  (public_name cmdliner)
- (flags :standard -w -3-6-27-32-35)
```
Comment (dbuenzli) [inline src/dune:3]: And that won't do it, or will it ?

File: dune (hunk 1 above)
Comment (dbuenzli) [reply dune:1]: The standard OCaml *warning* flags.

File: src/dune (hunk 2 above)
Comment (kit-ty-kate) [reply src/dune:3]: why? By default in dune `(flags :standard)` and the `(env ...)` stanza above simply changes the value of `:standard` for everything is the project. So it just works fine.

File: dune (hunk 1 above)
Comment (kit-ty-kate) [reply dune:1]: I don't really understand the request. The standard warning flag in the one used when the ocamlc/ocamlopt are called without setting `-w` so this is exactly what this is doing as it replaces the set of standard flags that dune has as its default to something that **does not** have `-w`

Comment (kit-ty-kate) [reply dune:1]: > We don't need to reflect ocamlbuild's `_tags` file the default build is `Makefile` based. We just need to reflect the standard OCaml flags.

in that case to match the flags set in `build.ml` we need:
```suggestion
(env (_ (flags -g -bin-annot -safe-string))) ; Use the same flags as with build.ml
```

**Comment by dbuenzli** [comment]
Thanks the patch is as 25b5c2159a2d10a802. Sorry for the misunderstanding!

**Pull request closed without merging**